src/main/scala/lms/core/stub.scala

```diff
@@ -803,37 +808,43 @@ object PrimitiveTypeLess {
 
   class NUM(override val x: Backend.Exp) extends TOP(x) {
     def +(y: NUM)(implicit pos: SourceContext): NUM = {
-      assert(t == y.t)
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t} in NUM + method")
       NUM(Adapter.g.reflect("+", x, y.x), t)
     }
     def -(y: NUM)(implicit pos: SourceContext): NUM = {
-      assert(t == y.t)
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t} in NUM - method")
       NUM(Adapter.g.reflect("-", x, y.x), t)
     }
     def *(y: NUM)(implicit pos: SourceContext): NUM = {
-      assert(t == y.t)
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t} in NUM * method")
       NUM(Adapter.g.reflect("*", x, y.x), t)
     }
     def /(y: NUM)(implicit pos: SourceContext): NUM = {
-      assert(t == y.t)
+      assert(t == y.t, s"t ${t} is not the same as y.t ${y.t} in NUM / method")
       NUM(Adapter.g.reflect("/", x, y.x),t)
     }
```

Comment:
these methods are still bad because if I have a (INT + INT) I get a NUM (instead of INT) as result. How to fix?